Development.md

```diff
@@ -142,6 +145,31 @@ clang -Xclang -ast-dump=json file.c
 
 Another option is to use clang-query ([tutorial](https://devblogs.microsoft.com/cppblog/exploring-clang-tooling-part-2-examining-the-clang-ast-with-clang-query/)).
 
+### Automated test case reduction
+
+In case of a crash, it may be possible to automatically reduce
+it using [C-Reduce](https://github.com/csmith-project/creduce).
+
+**Important:**
+On macOS, use `brew install --HEAD creduce`,
+as the default version is very outdated.
+
+There is a helper script [tools/reduce.py](/tools/reduce.py)
+which can coordinate `scip-clang` and `creduce`, since getting
+the details right is a bit finicky in the general case.
+
+It can be invoked like so:
+
+```bash
+# Pre-conditions:
+# 1. CWD is project root
+# 2. bad.json points to a compilation database with a single entry
+#    known to cause the crash
+/path/to/tools/reduce.py bad.json
```

Comment:
Noob q, but what's the format of the JSON for that?

Reply:
https://clang.llvm.org/docs/JSONCompilationDatabase.html

Reply:
Drop a link to that for posterity? Or maybe more usefully... what's the workflow for using this tool? It is implicit but you could make it explicit and it would be easier for people to get started with it.

Reply:
I don't think this is the right place for mentioning what a compilation database is; the first mention of it is in the "Running the indexer" section, so I'm assuming someone has already run the indexer before. I've added a link to the specification in that section.

I've also added a statement about what should happen after running the script.